packages/eui/src/components/code/code_block.test.tsx

```diff
@@ -174,40 +130,38 @@ describe('EuiCodeBlock', () => {
           {code}
         </EuiCodeBlock>
       );
-      expect(container.firstChild).toMatchSnapshot();
+
+      expect(container).toBeInTheDocument();
     });
 
-    describe('type checks', () => {
-      it('requires overflowHeight', () => {
-        // @ts-expect-error should expect overflowHeight
-        render(<EuiCodeBlock isVirtualized overflowHeight={undefined} />);
-      });
-
-      it('only allows whiteSpace of pre', () => {
-        render(
-          // @ts-expect-error should only accept "pre"
-          <EuiCodeBlock
-            isVirtualized
-            overflowHeight={50}
-            whiteSpace="pre-wrap"
-          />
-        );
-        // OK
-        render(
-          <EuiCodeBlock isVirtualized overflowHeight={50} whiteSpace="pre" />
-        );
-      });
+    it('requires overflowHeight', () => {
+      // @ts-expect-error should expect overflowHeight
+      render(<EuiCodeBlock isVirtualized overflowHeight={undefined} />);
+    });
+
+    it('only allows whiteSpace of pre', () => {
+      render(
+        // @ts-expect-error should only accept "pre"
+        <EuiCodeBlock isVirtualized overflowHeight={50} whiteSpace="pre-wrap" />
+      );
+
+      render(
+        <EuiCodeBlock isVirtualized overflowHeight={50} whiteSpace="pre" />
+      );
     });
   });
 
-  describe('line numbers', () => {
+  describe('Line numbers', () => {
     it('renders line numbers', () => {
       const { container } = render(
         <EuiCodeBlock lineNumbers {...requiredProps}>
           {code}
         </EuiCodeBlock>
       );
-      expect(container.firstChild).toMatchSnapshot();
+
+      expect(
+        container.querySelectorAll('.euiCodeBlock__lineNumberWrapper').length
```

Comment:
Let's target the specific number class instead of the wrapper
```suggestion
        container.querySelectorAll('.euiCodeBlock__lineNumber').length
```

Reply:
As above: https://github.com/elastic/eui/pull/8185#discussion_r1858403181

I'll try it again.